ui/marketplace/useWalletConnection.tsx

```diff
@@ -0,0 +1,39 @@
+import { useRouter } from 'next/router';
+import { useEffect, useRef } from 'react';
+
+import removeQueryParam from 'lib/router/removeQueryParam';
+import updateQueryParam from 'lib/router/updateQueryParam';
+import useWallet from 'ui/snippets/walletMenu/useWallet';
+
+export default function useMarketplaceWallet() {
```

Comment:
```suggestion
export default function useWalletConnection() {
```

Reply:
maybe I would rename it into `useConnectWalletAction`, `useAutoConnectWallet` or something similar